src/wallet/scriptpubkeyman.cpp

```diff
@@ -287,6 +287,23 @@ bool LegacyScriptPubKeyMan::TopUp(unsigned int size)
     return TopUpKeyPool(size);
 }
 
+void LegacyScriptPubKeyMan::MarkUnusedAddresses(const CScript& script)
+{
+    AssertLockHeld(cs_wallet);
+    // extract addresses and check if they match with an unused keypool key
+    for (const auto& keyid : GetAffectedKeys(script, *this)) {
+        std::map<CKeyID, int64_t>::const_iterator mi = m_pool_key_to_index.find(keyid);
+        if (mi != m_pool_key_to_index.end()) {
+            WalletLogPrintf("%s: Detected a used keypool key, mark all keypool key up to this key as used\n", __func__);
+            MarkReserveKeysAsUsed(mi->second);
+
+            if (!TopUpKeyPool()) {
```

Comment:
I'm also a bit confused about the distinction between `TopUpKeyPool()` and `Topup()`. In the native descriptor wallet PR there's an internal call to `Topup()` inside `DescriptorScriptPubKeyMan::GetNewDestination` and `DescriptorScriptPubKeyMan::MarkUnusedAddresses`: https://github.com/bitcoin/bitcoin/pull/16528/files#diff-5462ceb8a760a507152ab8b76bd48774R1459

Reply:
In `LegacyScriptPubKeyMan`, `TopUp()` and `TopUpKeyPool()` are the same. Since `TopUpKeyPool` was already being used throughout existing `CWallet` code that was moved into `LegacyScriptPubKeyMan`, I decided to let those calls just keep using `TopUpKeyPool`. But for the `ScriptPubKeyMan` interface itself, I wanted it to be named `TopUp` because that conceptually made more sense than calling it `TopUpKeyPool`.

Reply:
Any reason you would not want to rename TopUpKeyPool method to TopUp and call it everywhere instead of having two LegacyScriptPubKeyMan methods that do the same thing?